src/xpcc/io/iostream.hpp

```diff
@@ -460,7 +460,7 @@ class IOStream
 	 * @param	fmt		Format string
 	 */
 	IOStream&
-	printf(const char* fmt, ...);
+	printf(const char* fmt, ...)  __attribute__((format(printf, 2, 3)));
 
 	IOStream&
 	vprintf(const char *fmt, va_list vlist);
```

Comment:
If I've read the GCC doc correctly you cannot apply the attribute check to vprintf like functions the only thing you can do is checking the string for correctness. 
`For functions where the arguments are not available to be checked (such as vprintf), specify the third parameter as zero. In this case the compiler only checks the format string for consistency.` -gcc documentation
But I assume that this is better than nothing and will add this in a free minute.